src/main/java/frc/robot/subsystems/Limelight.java

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+/*
+ * Asimov's Laws:
+ * The First Law: A robot may not injure a human being or, through inaction, allow a human being to come to harm.
+ * The Second Law: A robot must obey the orders given it by human beings except where such orders would conflict with the First Law.
+ * The Third Law: A robot must protect its own existence as long as such protection does not conflict with the First or Second Law.
+ */
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.networktables.DoubleArraySubscriber;
+import edu.wpi.first.networktables.DoublePublisher;
+import edu.wpi.first.networktables.DoubleSubscriber;
+import edu.wpi.first.networktables.DoubleTopic;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+/** Subsystem for the Limelight 2+ that we use for vision. */
+public class Limelight extends SubsystemBase {
+
+  /* variable chaingun, I promise we use all of these */
+  private DoubleSubscriber tvSubscriber, txSubscriber, tySubscriber, taSubscriber;
+  /* See https://docs.limelightvision.io/en/latest/apriltags_in_3d.html#robot-localization-botpose-and-megatag
+   * to understand what the heck the different indices in this array mean
+   */
+  private DoubleArraySubscriber botposSubscriber;
+  private DoubleSubscriber pipelineSubcriber;
+  private DoublePublisher pipelinePublisher;
+  private String name;
+
+  private String fmtPath(String end) {
+    return "/" + name + "/" + end;
+  }
+
+  /**
+   * Creates a new Limelight.
+   *
+   * @param name The hostname of the Limelight
+   */
+  public Limelight(String name) {
+    this.name = name;
+    tvSubscriber = NetworkTableInstance.getDefault().getDoubleTopic(fmtPath("tv")).subscribe(0.0);
+    txSubscriber = NetworkTableInstance.getDefault().getDoubleTopic(fmtPath("tx")).subscribe(0.0);
+    tySubscriber = NetworkTableInstance.getDefault().getDoubleTopic(fmtPath("ty")).subscribe(0.0);
+    taSubscriber = NetworkTableInstance.getDefault().getDoubleTopic(fmtPath("ta")).subscribe(0.0);
+    DoubleTopic pipelineTopic =
+        NetworkTableInstance.getDefault().getDoubleTopic(fmtPath("pipeline"));
+    this.pipelineSubcriber = pipelineTopic.subscribe(0.0);
+    this.pipelinePublisher = pipelineTopic.publish();
+    double[] defaultBotpos = {0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
+    botposSubscriber =
+        NetworkTableInstance.getDefault()
+            .getDoubleArrayTopic(fmtPath("botpose"))
+            .subscribe(defaultBotpos);
+  }
+
+  /* now its time for getter method chaingun, which I have to write manually because VS Code */
```

Comment:
Capitalization. Also maybe remove the "write manually" "python script" parts if you want; keep if you value useless comedy.